optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/uni/BavetAbstractUniConstraintStream.java

```diff
@@ -144,133 +144,147 @@ public void addChildStream(BavetAbstractUniConstraintStream<Solution_, A> childS
-        childStreamList.add(stream);
-        return constraint;
+        return impactScoreConfigurableLong(constraintPackage, constraintName, matchWeigher, false);
     }
 
     @Override
     public Constraint penalizeConfigurableBigDecimal(String constraintPackage, String constraintName, Function<A, BigDecimal> matchWeigher) {
-        BavetConstraint<Solution_> constraint = buildConstraintConfigurable(constraintPackage, constraintName, false);
-        BavetScoringUniConstraintStream<Solution_, A> stream = new BavetScoringUniConstraintStream<>(constraintFactory, this, constraint, matchWeigher);
-        childStreamList.add(stream);
-        return constraint;
+        return impactScoreConfigurableBigDecimal(constraintPackage, constraintName, matchWeigher, false);
     }
 
     @Override
-    public Constraint reward(String constraintPackage, String constraintName, Score<?> constraintWeight) {
-        BavetConstraint<Solution_> constraint = buildConstraint(constraintPackage, constraintName, constraintWeight, true);
-        BavetScoringUniConstraintStream<Solution_, A> stream = new BavetScoringUniConstraintStream<>(constraintFactory, this, constraint);
+    public Constraint reward(String constraintPackage, String constraintName, Score<?> constraintWeight, ToIntFunction<A> matchWeigher) {
+        return impactScore(constraintPackage, constraintName, constraintWeight, matchWeigher, true);
+    }
+
+    @Override
+    public Constraint rewardLong(String constraintPackage, String constraintName, Score<?> constraintWeight, ToLongFunction<A> matchWeigher) {
+        return impactScoreLong(constraintPackage, constraintName, constraintWeight, matchWeigher, true);
+    }
+
+    @Override
+    public Constraint rewardBigDecimal(String constraintPackage, String constraintName, Score<?> constraintWeight, Function<A, BigDecimal> matchWeigher) {
+        return impactScoreBigDecimal(constraintPackage, constraintName, constraintWeight, matchWeigher, true);
+    }
+
+    @Override
+    public Constraint rewardConfigurable(String constraintPackage, String constraintName, ToIntFunction<A> matchWeigher) {
+        return impactScoreConfigurable(constraintPackage, constraintName, matchWeigher, true);
+    }
+
+    @Override
+    public Constraint rewardConfigurableLong(String constraintPackage, String constraintName, ToLongFunction<A> matchWeigher) {
+        return impactScoreConfigurableLong(constraintPackage, constraintName, matchWeigher, true);
+    }
+
+    @Override
+    public Constraint rewardConfigurableBigDecimal(String constraintPackage, String constraintName, Function<A, BigDecimal> matchWeigher) {
+        return impactScoreConfigurableBigDecimal(constraintPackage, constraintName, matchWeigher, true);
+    }
+
+    @Override
+    protected Constraint impactScore(String constraintPackage, String constraintName, Score<?> constraintWeight,
+            boolean positively) {
```

Comment:
Consistency: boolean "postively" has a different name than boolean "positive" in
   buildConstraintConfigurable(String constraintPackage, String constraintName, boolean positive) {

They should have the same name, because they are the same semantic.